source/server/active_tcp_socket.h

```diff
@@ -83,8 +83,7 @@ struct ActiveTcpSocket : public Network::ListenerFilterManager,
 
   // Network::ListenerFilterCallbacks
   Network::ConnectionSocket& socket() override { return *socket_.get(); }
-  Event::Dispatcher& dispatcher() override;
-  void continueFilterChain(bool success) override;
+  void continueFilterChain(bool success);
```

Comment:
Likewise comments in https://github.com/envoyproxy/envoy/pull/22732/files/e8563f3153394c2ffe8653ca9c9beef7e70dacc0#diff-6708370e112a5097f06e16009c3730b34f5ab0ee4f1e736f8816848409435888R122

and https://github.com/envoyproxy/envoy/blob/fb2d0a059d25d96937b7efcd9ad4d5e3a415eaab/envoy/network/filter.h#L333

should be updated since the convention is changing e.g the filter doesn't call continue

Reply:
> Can we move this elsewhere since it's no longer part of the interface commented above and have it documented?
The ActiveTcpSocket is a struct, its all members are public now, I guess that is for convenient for ActiveTcpListener access. And ActiveTcpListener is only place ActiveTcpSocket, and probably we think ActiveTcpSocket is kind of internal struct for ActiveTcpListener(since its only alive short time, it will be deleted when the connection passed all the listener filters or the connection rejected by listener filter)

> It seems like it's now split into either internal usage or external `continueFilterChain(true)` call that really just starts the filter chain. We could increase encapsulation splitting this out into startFilterChain() and continueFilterChain with the latter being private and startFilterChain just deferring to it.

Yea, that is good idea! The ActiveTcpListener is the place to call `startFileChain()`, and that make the code clear. Just as above said the whole ActiveTcpSocket are public, not sure I should make continueFilterChain as private.

Reply:
> Likewise comments in https://github.com/envoyproxy/envoy/pull/22732/files/e8563f3153394c2ffe8653ca9c9beef7e70dacc0#diff-6708370e112a5097f06e16009c3730b34f5ab0ee4f1e736f8816848409435888R122
> 
> and
> 
> https://github.com/envoyproxy/envoy/blob/fb2d0a059d25d96937b7efcd9ad4d5e3a415eaab/envoy/network/filter.h#L333
> 
> should be updated since the convention is changing e.g the filter doesn't call continue

got it, will update the comments

Reply:
Thanks this looks great! It makes a lot of sense that a lot of the class was open as you said, but it seems like it has grown in responsibility e.g. internally buffering listener data. 

Having some methods private (`continueFilterChain`, `createListenerFilterBuffer`) would make the interface more compact / easier to grok.

Reply:
got it, let me try to improve that.